### PR TITLE
[RLlib] Fix bug in SingleAgentEnvRunner: Calling `sample()` would always force-reset the vector env (even if episodes were not completed in a previous call).

### PR DESCRIPTION
**Pull request by sven1977**
Fix bug in SingleAgentEnvRunner: Calling `sample()` repeatedly would always force-reset the vector env (even if episodes were not completed in a previous call).

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/ppo/ppo.py [hunk 1]
```diff
@@ -439,8 +439,28 @@ def training_step(self) -> ResultDict:
...
                     )
+                # TEST
+                #import pickle
+                #import torch
+                #import os
+                #with open("train_batch.pkl", "wb") as file:
+                #    pickle.dump(train_batch, file)
+                #torch.save(self.workers.local_worker().policy_map["default_policy"].model.state_dict(), "model_weights.pth")
+                # END TEST
+
```
Comment (kouroshHakha) [inline rllib/algorithms/ppo/ppo.py:450]: remove?

File: rllib/algorithms/ppo/ppo.py [hunk 2]
```diff
@@ -439,8 +439,28 @@ def training_step(self) -> ResultDict:
...
+                #import pickle
+                #import torch
+                #import os
+                #from ray.rllib.utils.test_utils import check
+                #rw_path = "/Users/sven/ray_results/PPO_2023-11-15_13-28-16/PPO_CartPoleDebug_74001_00000_0_2023-11-15_13-28-16"
+                #with open(os.path.join(rw_path, "train_batch.pkl"), "rb") as file:
+                #    rw_train_batch = pickle.load(file)
+                #rw_state_dict = torch.load(os.path.join(rw_path, "model_weights.pth"))
+                #self.workers.local_worker().module.load_state_dict(rw_state_dict)
+                # END TEST
```
Comment (kouroshHakha) [inline rllib/algorithms/ppo/ppo.py:463]: remove?

File: rllib/algorithms/ppo/ppo.py [hunk 3]
```diff
@@ -457,6 +477,9 @@ def training_step(self) -> ResultDict:
                 train_batch: SampleBatch = postprocess_episodes_to_sample_batch(
                     postprocessed_episodes
                 )
+                #check(train_batch["advantages"], rw_train_batch["default_policy"]["advantages"], rtol=0.000001)
+                #check(train_batch["vf_preds"], rw_train_batch["default_policy"]["vf_preds"])
+                #check(train_batch["value_targets"], rw_train_batch["default_policy"]["value_targets"])
```
Comment (kouroshHakha) [inline rllib/algorithms/ppo/ppo.py:482]: remove?

**Review by kouroshHakha — APPROVED** [review]

File: rllib/algorithms/ppo/ppo.py (hunk 1 above)
Comment (sven1977) [reply rllib/algorithms/ppo/ppo.py:450]: Sorry, will comb through changes and clean everything up before merging. Thanks for this catch!

Comment (sven1977) [reply rllib/algorithms/ppo/ppo.py]: done

File: rllib/algorithms/ppo/ppo.py (hunk 2 above)
Comment (sven1977) [reply rllib/algorithms/ppo/ppo.py]: done

**Pull request merged**